### PR TITLE
refactor: use a more reactive approach for handling signer state changes

### PR DESCRIPTION
**Pull request by moldy530**

### CODE REVIEW

**Comment by moldy530** [comment]
NOTE: we need to merge, publish, and reference https://github.com/alchemyplatform/aa-sdk/pull/523 in this PR before merging

File: env.mjs [hunk 1]
```diff
@@ -19,7 +19,11 @@ export const env = createEnv({
    * isn't built with invalid env vars. To expose them to the client, prefix them with
    * `NEXT_PUBLIC_`.
    */
-  client: {},
+  client: {
+    NEXT_PUBLIC_ROOT_ORG_ID: z
+      .string()
+      .default("3121a8a0-c548-4d14-a313-630c3b739858"),
```
Comment (avasisht23) [inline env.mjs:25]: should we expose this?

File: app/providers.tsx [hunk 2]
```diff
@@ -8,14 +10,18 @@ const TurnkeyIframeElementId = "turnkey-iframe-element-id";
...
+  >({
     connection: {
       rpcUrl: "/api/rpc",
     },
     iframeConfig: {
       iframeContainerId: TurnkeyIframeContainerId,
       iframeElementId: TurnkeyIframeElementId,
     },
+    rootOrgId:
+      typeof window !== "undefined" ? env.NEXT_PUBLIC_ROOT_ORG_ID : undefined,
```
Comment (avasisht23) [inline app/providers.tsx:24]: learn: why is this needed again? for mobile?

Comment (moldy530) [reply app/providers.tsx:24]: this is only relevant in an SSR context and not used on the server

File: env.mjs (hunk 1 above)
Comment (moldy530) [reply env.mjs:25]: yea there's nothing wrong with exposing this imo. it gets used either way in production for certain whoamis

File: app/providers.tsx (hunk 2 above)
Comment (moldy530) [reply app/providers.tsx:24]: oh and the rootOrgId is used for us to be able to test against staging mainly. It's not really meant to be used by other entities and probably won't even work for them

**Review by avasisht23 — APPROVED** [review]

**Comment by moldy530** [comment]
closing this out in favor of a PR that will build on https://github.com/alchemyplatform/aa-sdk/pull/539
